app/scripts/controllers/transactions/index.js

```diff
@@ -727,6 +718,24 @@ export default class TransactionController extends EventEmitter {
       this.txStateManager.setTxStatusConfirmed(txId);
       this._markNonceDuplicatesDropped(txId);
 
+      const { submittedTime } = txMeta;
+      const { blockNumber } = txReceipt;
+      const metricsParams = { gas_used: gasUsed };
+      const completionTime = await this._getTransactionCompletionTime(
+        blockNumber,
+        submittedTime,
+      );
+
+      if (completionTime) {
+        metricsParams.completion_time = completionTime;
+      }
+
```

Comment:
Lets do one more clause here
```suggestion
if (txReceipt.status === '0x0') {
  metricsParams.status = 'failed on-chain'; ((overriding the status we set in that event))
  // metricsParams.error = TODO: figure out a way to get the on-chain failure reason
}
```

Reply:
done